guides/managing-current-user.md

```diff
@@ -72,6 +72,50 @@ export default Ember.Service.extend({
 });
 ```
 
+#### Using a dedicated endpoint and returning actual user id in response
+
+```js
+// app/services/current-user.js
+import Ember from 'ember';
+
+const { inject: { service }, isEmpty, RSVP } = Ember;
+
+export default Ember.Service.extend({
+  session: service('session'),
+  store: service(),
+
+  load() {
+    if (this.get('session.isAuthenticated')) {
+      return this.get('store').queryRecord('user', { me: true }).then((user) => {
+        this.set('user', user);
+      });
+    }
+  }
+});
+
+// app/adapters/user.js
+import ApplicationAdapter from './application';
+
+export default ApplicationAdapter.extend({
+  urlForQueryRecord(query) {
+    if (query.me) {
+      delete query.me;
+      return `${this._super(...arguments)}/me`;
+    }
+
+    return this._super(...arguments);
+  }
+});
+```
+
+If the `find` method is called with id `me` and the response includes the actual
+user id, ember data will create an empty record with id `me` and all other model
+attributes empty.
```

Comment:
`find` is actually private, I think we should use `store.findRecord` here as it is more idiomatic ember-data use, even though it has been mentioned before.

I am not sure we should start here why `findRecord('user', 'me')` is the wrong choice and instead say why we are using `store.queryRecord()`: because it's the way to fetch a record in ember data when the `id` is not known beforehand.

And then maybe outline with a `Note: ` why we are not using `findRecord("user", "me")`: because ember data expects the returned model to have the same id, as otherwise an unused empty record with `id` of `me` is in the store.